simplyblock_core/services/spdk_stats_collector.py

```diff
@@ -1,37 +1,45 @@
-import datetime
+from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
 import time
-
 from simplyblock_core.services.spdk import client as spdk_client
-
-
-method = "thread_get_stats"
-spdk_sock_path = '/var/tmp/spdk.sock'
-csv_file_path = "/etc/simplyblock/spdk_top.csv"
-
-def get_file(ret):
-
-    f = open(csv_file_path, 'at', buffering=1)
-    f.write("date, ")
-    for stats in ret["threads"]:
-        f.write(", ".join([k for k in stats]))
-    f.write("\n")
-    return f
-
-
-if __name__ == '__main__':
-
-    client = spdk_client.JSONRPCClient(spdk_sock_path, 5260)
-
-    out_file = None
+import socket
+
+PUSHGATEWAY_URL = "http://pushgateway:9091"
+SPDK_SOCK_PATH = "/var/tmp/spdk.sock"
+
+def get_mgmt_ip():
+    return socket.gethostbyname(socket.gethostname())
```

Comment:
Since this code is expected to also work in on-prem environment, it could also be possible there are multiple network interfaces. How do we know which network interface to use?

Reply:
you clearly know the interface name, them maybe something like this can be used to get the exact private address associated with the interface on the local machine. 
https://stackoverflow.com/a/24196955